### ISSUE TITLE
Adding this plugin results in MissingPluginException for Firebase Core

### ISSUE THREAD

**Issue by cemozerr** [mwaylabs/flutter-apns#37]
Adding this plugin to the pubspec.yaml (without even using or importing it anywhere), results in the exception below: 
```
[VERBOSE-2:ui_dart_state.cc(177)] Unhandled Exception: MissingPluginException(No implementation found for method Firebase#initializeCore on channel plugins.flutter.io/firebase_core)
#0      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:157:7)
<asynchronous suspension>
#1      MethodChannel.invokeMethod (package:flutter/src/services/platform_channel.dart:332:12)
#2      MethodChannel.invokeListMethod (package:flutter/src/services/platform_channel.dart:345:41)
#3      MethodChannelFirebase._initializeCore (package:firebase_core_platform_interface/src/method_channel/method_channel_firebase.dart:30:36)
#4      MethodChannelFirebase.initializeApp (package:firebase_core_platform_interface/src/method_channel/method_channel_firebase.dart:75:13)
#5      Firebase.initializeApp (package:firebase_core/src/firebase.dart:43:25)
#6      main (package:mobile/main.dart:14:18)
#7      _runMainZoned.<anonymous closure>.<anonymous closure> (dart:ui/hooks.dart:231:25)
#8      _rootRun (d<…>
```

In this project where the error occurs, I also import and use `firebase_core` and `firebase_auth`. The error occurs at the `main()` function in `main.dart` where I have the code below before calling the `runApp()` method on my app.
```
  WidgetsFlutterBinding.ensureInitialized();
  await Firebase.initializeApp(); // <-- Error occurs here
```

I've tried your `1.4.0`, `1.3.1`, `1.3.0` and `1.0.5` releases to remedy the issue, and even created a new flutter project with the same `lib/` and `pubspec.yaml` to no avail.

The issue occurs with iOS, haven't tried with Android.

**Comment by cemozerr** [comment]
I believe the issue here is due to the line of code in https://github.com/mwaylabs/flutter-apns/blob/master/ios/Classes/FlutterApnsSwizzler.m#L42:
`[self disablePluginNamed:@"FLTFirebaseCorePlugin"];`

Not sure why you need to disable the FirebaseCorePlugin, but my app definitely does not load the FirebaseCore plugin afterward and thus simply fails when it needs to call a method from it.

**Comment by cemozerr** [comment]
I was able to get notifications in my app working by commenting out the "disable Firebase core plugin line" I mentioned in the comment above and also by commenting out the `fatalError` being thrown when the `didSwizzle` method returns false here: 
https://github.com/mwaylabs/flutter-apns/blob/master/ios/Classes/FlutterApnsPlugin.swift#L22

And yes this is a hack, but I'm running out of ideas. Hopefully, this information will be helpful if there is a bug in `flutter_apns`.

**Comment by cemozerr** [comment]
Turns out I was only able to make the iOS Simulator work in this way. Not when the app is running on my phone.

**Comment by szotp** [comment]
Hi, I did not expect people already using firebase on iOS to be interested in this plugin, that's why it disables the core. Your hacks should work though, I may add support for them in the plugin.

For running on device, have you seen https://github.com/mwaylabs/flutter-apns#troubleshooting ?

**Comment by cakkiraz** [comment]
> I believe the issue here is due to the line of code in https://github.com/mwaylabs/flutter-apns/blob/master/ios/Classes/FlutterApnsSwizzler.m#L42:
> `[self disablePluginNamed:@"FLTFirebaseCorePlugin"];`
> 
> Not sure why you need to disable the FirebaseCorePlugin, but my app definitely does not load the FirebaseCore plugin afterward (which it needs for firebase auth) and thus simply fails when it needs to call a method from it.
> 
> Is this APNS plugin simply not supposed to work on an app that utilizes Firebase Core?

Hi, I've been suffering from the same problem for a while. I have found your solution. But I don't know how can I edit this package to handles this problem? Can you help me? Thank you so much :)

**Comment by cakkiraz** [comment]
I am suffering from the same problem :( I was using firebase packages. I need also this package. How can I modify this package to use with firebase?

Thank you...

**Comment by szotp** [comment]
Sure I will implement something. But I'm curious - why not just use firebase_messaging?

**Comment by cemozerr** [comment]
I, for example, use a chat provider called get-stream for chat, and firebase auth for authentication in my app. 

Get-stream delivers chat notification through apns for iOS and through firebase-messaging for android. Thus, to easily integrate with my chat-provider's push notifications, I need both apns and firebase-messaging.

**Comment by cemozerr** [comment]
> I am suffering from the same problem :( I was using firebase packages. I need also this package. How can I modify this package to use with firebase?
> 
> 
> 
> Thank you...

Shoot me an email @cakkiraz and I can try to help. 👍

**Comment by cakkiraz** [comment]
Thank you so much @szotp and @cemozerr. :)

I will send you mail @cemozerr 🙏🏻

**Comment by ujnamss** [comment]
Hi @cemozerr & @szotp - I am also encountering the same issue. My use case is almost exactly same as that of @cemozerr. I use the core firebase_messaging for android and want to use flutter_apns to handle iOS. A fix would be super helpful.

@cemozerr - in the mean time am happy to send a note to you directly, if you could help that will be awesome!

**Comment by shashikantdurge** [comment]
Had to switch to other plugin apns_flutter due to this issue. Please resolve it asap

**Comment by szotp** [comment]
Implemented:
https://github.com/mwaylabs/flutter-apns#enabling-firebasecore

I'll release new version soon.

**Comment by szotp** [comment]
Implemented in 1.4.1

**Comment by aytunch** [comment]
Just like @cakkiraz I need to use Firebase Messaging and also I need to use a 3rd party Chat service which will be relying on the APNS token instead of the FCM token. I asked over at Firebase Messaging if they plan on supporting APNS tokens to trigger onMessage but no answer yet.
> I, for example, use a chat provider called get-stream for chat, and firebase auth for authentication in my app.
> 
> Get-stream delivers chat notification through apns for iOS and through firebase-messaging for android. Thus, to easily integrate with my chat-provider's push notifications, I need both apns and firebase-messaging.

Version 1.4.1 states that;
> If you want to use firebase, but not firebase messaging, add this configuration entry in your Info.plist:
> 
> <key>flutter_apns.disable_firebase_core</key>
> <false/>
> 

Does this mean we can use Firebase services except Firebase Messaging? Just like @cemozerr, @ujnamss and @shashikantdurge stated, we need Flutter_Apns working together with Firebase_messaging. Please clarify if the documentation above is misleading, or if I misunderstood version 1.4.1.

Thanks people.
